include/nuttx/progmem.h

```diff
@@ -49,96 +49,127 @@ extern "C"
  * Public Functions Definitions
  ****************************************************************************/
 
+/* NOTE: Since vendors use the terms page, block, and sector interchangably,
+ * it causes confusion and therefore a neutral term is used. We use the term
+ * "index" to refer to the ordinal number of erasable units of memory.
+ */
+
 /****************************************************************************
- * Name: up_progmem_neraseblocks
+ * Name: up_progmem_maxeraseindex
  *
  * Description:
- *   Return number of erase blocks
+ *   Return the total number of erasable units of memory.
+ *
+ * Returned Value:
+ *   Returns number of erasable units of memory, or 0 if there are no units
+ *   available that can be programmed.
  *
  ****************************************************************************/
 
-size_t up_progmem_neraseblocks(void);
+size_t up_progmem_maxeraseindex(void);
 
 /****************************************************************************
  * Name: up_progmem_isuniform
  *
  * Description:
- *  Is program memory uniform or erase page and read/write page size differs?
+ *  Are all erasable units of memory the same size?
```

Comment:
```suggestion
 *   Are all erasable units of memory the same size?
```
nit: Uniformize alignment of description text.